File: baekjoon/10211.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    freopen("input.txt", "r", stdin);
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        int arr[1002];
        int pSum[1002];
        for(int i=0; i<n; i++){
            int tmp;
            cin>>tmp;
            arr[i]=tmp;
            if(i==0){
                pSum[0]=tmp;
            }
            else{
                pSum[i]=pSum[i-1]+tmp;
            }
        }
        int max_val=INT_MIN;
        for(int i=0; i<n; i++){
            for(int j=i; j<n; j++){
                max_val = max(max_val, pSum[j]-pSum[i]+arr[i]);
            }
        }
        cout<<max_val<<'\n';
    }
    return 0;
};

File: baekjoon/10211.py
import sys
import math
sys.stdin = open('input.txt')

t = int(sys.stdin.readline())
while t:
    t-=1
    n = int(sys.stdin.readline())
    lis = list(map(int, sys.stdin.readline().split()))
    pSum=[]
    cnt=0
    for item in lis:
        cnt+=item
        pSum.append(cnt)
    max_val=-math.inf
    for i in range(len(pSum)):
        for j in range(i, len(pSum)):
            max_val = max(pSum[j]-pSum[i]+lis[i], max_val)
    sys.stdout.write(str(max_val)+'\n')
    # print(max_val)